config-controller/internal/controller/policy_controller.go

```diff
@@ -18,45 +18,42 @@ package controller
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 
 	"github.com/pkg/errors"
 	configstackroxiov1alpha1 "github.com/stackrox/rox/config-controller/api/v1alpha1"
-	v1 "github.com/stackrox/rox/generated/api/v1"
-	storage "github.com/stackrox/rox/generated/storage"
-	"github.com/stackrox/rox/roxctl/common"
-	"github.com/stackrox/rox/roxctl/common/auth"
-	roxctlIO "github.com/stackrox/rox/roxctl/common/io"
-	"github.com/stackrox/rox/roxctl/common/logger"
-	"github.com/stackrox/rox/roxctl/common/printer"
-	"google.golang.org/grpc"
-	"google.golang.org/protobuf/encoding/protojson"
-	"google.golang.org/protobuf/protoadapt"
+	"github.com/stackrox/rox/config-controller/pkg/client"
+	"github.com/stackrox/rox/generated/storage"
 	k8serr "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 	ctrlClient "sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
-// PolicyReconciler reconciles a Policy object
-type PolicyReconciler struct {
-	ctrlClient.Client
-	Scheme *runtime.Scheme
-	conn   *grpc.ClientConn
+// SecurityPolicyReconciler reconciles a SecurityPolicy object
+type SecurityPolicyReconciler struct {
+	K8sClient    ctrlClient.Client
+	Scheme       *runtime.Scheme
+	PolicyClient client.CachedPolicyClient
 }
 
 //+kubebuilder:rbac:groups=config.stackrox.io,resources=policies,verbs=get;list;watch;create;update;patch;delete
 //+kubebuilder:rbac:groups=config.stackrox.io,resources=policies/status,verbs=get;update;patch
 //+kubebuilder:rbac:groups=config.stackrox.io,resources=policies/finalizers,verbs=update
 
-func (r *PolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+func (r *SecurityPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	rlog := log.FromContext(ctx)
 	rlog.Info("Reconciling", "namespace", req.Namespace, "name", req.Name)
 
+	if err := r.PolicyClient.EnsureFresh(ctx); err != nil {
+		return ctrl.Result{}, errors.Wrap(err, "Failed to refresh")
+	}
+
 	// Get the policy CR
-	policyCR := &configstackroxiov1alpha1.Policy{}
-	if err := r.Client.Get(ctx, req.NamespacedName, policyCR); err != nil {
+	policyCR := &configstackroxiov1alpha1.SecurityPolicy{}
+	if err := r.K8sClient.Get(ctx, req.NamespacedName, policyCR); err != nil {
```

Comment:
Comment from @clickboo on #12562: We could log a warning if expected CR is not found

Reply:
This could be a normal scenario in K8s.  Thinking about this, though, we should add a finalizer to policy CRs to ensure they don't get removed before we have a chance to ensure the policy is deleted from Central.